Core/clim-basic/text-formatting.lisp

```diff
@@ -153,34 +153,65 @@
+(defgeneric invoke-with-filling-output
+    (stream continuation
+     &key fill-width break-characters
+       after-line-break after-line-break-initially after-line-break-subsequent)
+  (:method ((stream filling-output-mixin) continuation
+            &key (fill-width '(80 :character))
+              break-characters
+              (after-line-break nil)
+              (after-line-break-initially nil)
+              (after-line-break-subsequent t))
+    (with-temporary-margins (stream :right `(:absolute ,fill-width))
+      (letf (((stream-end-of-line-action stream) :wrap*)
+             ((line-break-strategy stream) break-characters))
+        (if (not after-line-break)
+            (funcall continuation stream)
+            (letf (((after-line-break stream) after-line-break)
+                   ((after-line-break-initially stream) after-line-break-initially)
+                   ((after-line-break-subsequent stream) after-line-break-subsequent)
+                   ((graphics-state-ink (graphics-state stream)) (medium-ink stream))
+                   ((graphics-state-text-style (graphics-state stream)) (medium-text-style stream)))
+              (funcall continuation stream)))))))
+
+(defmacro filling-output ((stream &rest args
+                                  &key
+                                  fill-width
                                   break-characters
 				  after-line-break
                                   after-line-break-initially
-                                  (after-line-break-subsequent t))
+                                  after-line-break-subsequent)
 			  &body body)
+  (declare (ignore fill-width break-characters after-line-break-subsequent))
   (setq stream (stream-designator-symbol stream '*standard-output*))
-  `(with-temporary-margins (,stream :right (list :absolute ,fill-width))
-     (letf (((stream-end-of-line-action ,stream) :wrap*)
-            ((line-break-strategy ,stream) ,break-characters)
-            ,@(when after-line-break
-                `(((after-line-break ,stream) ,after-line-break)
-                  ((after-line-break-initially ,stream) ,after-line-break-initially)
-                  ((after-line-break-subsequent ,stream) ,after-line-break-subsequent)
-                  ((graphics-state-ink (graphics-state ,stream)) (medium-ink ,stream))
-                  ((graphics-state-text-style (graphics-state ,stream)) (medium-text-style ,stream)))))
-       ,(when (and after-line-break after-line-break-initially)
-          `(etypecase (after-line-break ,stream)
-             (string   (write-string (after-line-break ,stream) ,stream))
-             (function (funcall (after-line-break ,stream) ,stream nil))))
-       ,@body)))
-
-(defmacro indenting-output ((stream indent &key (move-cursor t)) &body body)
+  (with-gensyms (continuation)
+    `(flet ((,continuation (,stream)
+              ,(when (and after-line-break after-line-break-initially)
```

Comment:
This is intended to insert the code if both keyword arguments are supplied, right? Their values (which are forms in the macro context) are irrelevant.

If so, this could be communicated more clearly by binding `-supplied-p` variables for the keyword parameters.

Also, is `after-line-break-initially` a Boolean or a function? If it is a Boolean, this compile-decision based on its value seems wrong.

Reply:
you are right, addressed